webrender/src/prim_store.rs

```diff
@@ -393,6 +400,10 @@ impl GradientData {
             ColorF::new(0.0, 0.0, 0.0, 0.0)
         };
 
+        // Fill in the first gradient entry with the first color stop
+        self.fill_colors(cur_idx, cur_idx + 1, &cur_color, &cur_color);
```

Comment:
this is a bit confusing. Perhaps, do `fill_colors(0, 1, ...)` followed by `cur_idx = 1;`?